app/common/errors.py

```diff
@@ -104,3 +104,34 @@ def __init__(self):
         code = 100
 
         super(ApiError, self).__init__(type, message, status, code, None)
+
+
+class UnauthenticatedRequestError(_BaseError):
+    """
+    An error indicating that the client did not provide a request
+    that could not be authenticated
+    """
+
+    def __init__(self, message, source):
+        type = 'UnauthenticatedRequestError'
+        if not message:
+            message = 'An authentication error has occurred'
```

Comment:
Let's make the message 'The requested resource cannot be accessed with the provided credentials'